core/src/main/java/org/elasticsearch/rest/RestController.java

```diff
@@ -215,6 +225,65 @@ void dispatchRequest(final RestRequest request, final RestChannel channel, final
     }
 
     /**
+     * If a request contains content, this method will return {@code true} if the {@code Content-Type} header is present, matches an
+     * {@link XContentType} or the request is plain text, and content type is required. If content type is not required then this method
+     * returns true unless a content type could not be inferred from the body and the rest handler does not support plain text
+     */
+    boolean checkForContentType(final RestRequest restRequest, final RestChannel channel, final int contentLength,
+                                final RestHandler restHandler) {
+        if (contentLength > 0) {
+            if (restRequest.getXContentType() == null) {
+                if (restHandler != null && restHandler.supportsPlainText()) {
+                    deprecationLogger.deprecated("Plain text request bodies are deprecated. Use request parameters or body " +
+                        "in a supported format.");
```

Comment:
that means the content-type is plain text and the handler must support plain text requests